specs/withdrawals.md

```diff
@@ -0,0 +1,204 @@
+
+1. For each `WithdrawalMessage` initiated on L2 (ie. with a unique `messageNonce`), the following properties must hold:
+    1. There should be one (and only one) relayable message.
+    1. It should not be possible to relay the message with any of its fields modified, ie.
+        1. Modifying the `sender` field would enable a 'spoofing' attack.
+        1. Modifying the `target`, `message`, or `value` fields would enable an attacker to dangerously change the
+           intended outcome of the withdrawal.
+        1. Modifying the `gasLimit` could make the cost of relaying too high, or allow the relayer to cause execution
+           to fail (out of gas) in the `target`.
+
+### Handling Successfully Verified Messages That Fail When Relayed
+
+If the execution of the relayed call fails in the `target` contracts, it is unfortunately not possible to determine
+whether or not it was 'supposed' to fail, and whether or not it should be 'replayable'.
+Thus we provide the following mitigations:
+
+// TODO: finish this sentence...
+
+1. The minimum gas amount to be
+
+[Insufficient Gas Griefing]:(https://swcregistry.io/docs/SWC-126)
+
+## Summary of Definitions
+
+### Constants
+
+| Name                  | Value     | Unit    |
+| --------------------- | --------- | ------- |
+| `FINALIZATION_WINDOW` | `604_800` | seconds |
+
+This `FINALIZATION_WINDOW` value is equivalent to 7 days.
+
+### Data Structures and Type Aliases
+
+> TODO: update this section is quite outdated with current impl
+
+1. A `WithdrawalMessage` is encoded in a struct as follows:
+
+    ```js
+    struct WithdrawalMessage {
+        uint256 nonce;
+        address sender;
+        uint256 value;
+        bytes message;
+    }
+    ```
+
+1. The `L2OutputTimestamp` is an alias for `uint256`, and MUST be a multiple of the `SUBMISSION_INTERVAL` described
+  in the [L2 Output](./proposals.md#constants) document.
+
+1. The `WithdrawalsRoot` is an alias for a `bytes32` value, corresponding to the [MPT][g-mpt]
+  storage root of the Withdrawals predeploy contract at `0x4200000000000000000000000000000000000015` (described below).
+
+1. The `WithdrawalsRootInclusionProof` proof contains the data necessary to prove that the provided `WithdrawalsRoot` is
+  included in the
+  [SSZ merkleization](https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md#merkleization)
+  of the `L2Output` as defined in the
+  [L2 output commitment construction](./proposals.md#l2-output-commitment-construction).
+
+1. A `WithdrawalMessageInclusionProof` is an MPT proof encoded in a struct as follows:
+
+   ```js
+   struct WithdrawalMessageInclusionProof {
+       WithdrawalMessage message;
+       bytes32 l2withdrawalsRoot;
+       bytes memory _key;      // storage key of the withdrawal message commitment
+       bytes memory _value;    // Always bytes32(1) (boolean validity status of the message)
```

Comment:
I feel like we don't need to include a constant in this struct

Reply:
yeah, specs need updating, it's not in the impl.